cmake/Obs2Ioda_CompilerFlags.cmake

```diff
@@ -0,0 +1,31 @@
+# Set Fortran compiler flags specific to the GNU Compiler
+# -cpp: Enable the C preprocessor
+# -ffree-line-length-none: Remove the limit on the length of lines in the source file
+# -mcmodel=medium: Allow for larger datasets in memory
+set(FORTRAN_COMPILER_GNU_FLAGS
+    $<$<COMPILE_LANGUAGE:Fortran>:-cpp -ffree-line-length-none -mcmodel=medium>
+)
+
+# Set Debugging Fortran compiler flags specific to the GNU Compiler
+# -fbacktrace: Provide a backtrace when an error occurs
+# -ffpe-trap=invalid,zero,overflow: Trap floating point exceptions (invalid calculation, divide by zero, overflow)
+# -fcheck=all: Execute all types of runtime checks
+set(FORTRAN_COMPILER_GNU_DEBUG_FLAGS
+    $<$<COMPILE_LANGUAGE:Fortran>:-fbacktrace -ffpe-trap=invalid,zero,overflow -fcheck=all>
```

Comment:
Is `-g` needed here as well?

Reply:
Generally CMake will add -g when the build type is set to Debug, but I will go ahead and add -g if this changes in future versions of CMake. Thanks for pointing this out!

Reply:
I suspected that CMake might add `-g` for debug builds, but then I noticed that `-g` was being explicitly added to `FORTRAN_COMPILER_INTEL_DEBUG_FLAGS` as part of this PR.